src/middlewared/middlewared/plugins/network.py

```diff
@@ -210,6 +227,14 @@ async def do_update(self, data):
         is_ha = True
         sync_group_mappings = False
 
+        # Apply on-disk settings and user provided ones to our
+        # defaults to ensure that the JSON stored in db
```

Comment:
If there is potential that the db isn't normalized, we should be handling that via a migration ?

Reply:
Well, the JSON is eventually written as a string in the db which means that it's not quite as robust as if things were properly typed. So me being a bit paranoid. We can change this in bluefin, but I'd rather not in 13. (This is backport from SCALE BTW).